src/rpdk/core/contract/suite/handler_delete.py

```diff
@@ -42,8 +42,12 @@ def deleted_resource(resource_client):
     finally:
         status, response = resource_client.call(Action.DELETE, model)
 
-        # a failed status is allowed if the error code is NotFound
-        if status == OperationStatus.FAILED:
+        # a failed status is allowed if the error code is NotFound for
+        # resource having read-only primary identifier
+        if (
+            not resource_client.has_writable_identifier()
```

Comment:
could you plz clarify this a bit `However, for resources which have writeable identifiers, this is not the case.`? why writable is different from non writable id?

Reply:
the delete fails with different error codes for this case

Reply:
how come

Reply:
hey Anton, for resources which have readOnly properties which are primaryIdentifiers the final delete fails with different error codes depending on the service because, the readOnly property is null. The service does not null primary identifier to be passed. We can not expect them to throw NotFound error code in this case.

Reply:
I think there is a little bit of misunderstanding. The condityion should be following:
```
if (
           resource_client.has_writable_identifier() and status == OperationStatus.FAILED
):